### ISSUE TITLE
Test failed: tracing/eventpipe/eventsvalidation/GCEvents/GCEvents.sh

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#26590]
**Job:**
[coreclr-gcstress-extra 20190907.1](https://dev.azure.com/dnceng/public/_build/results?buildId=343741)

**Detail:**
Linux x64: https://helix.dot.net/api/2019-06-17/jobs/95ccd71a-b8b9-444c-976e-d0058df23782/workitems/PayloadGroup0/console
windows x64: https://helix.dot.net/api/2019-06-17/jobs/71500183-fcb6-48db-9fb9-ab9083750592/workitems/PayloadGroup0/console
Windows x86: https://helix.dot.net/api/2019-06-17/jobs/b6387512-4245-4192-a9e5-4a185314459d/workitems/PayloadGroup0/console

**OS & Arch:**
Windows x64
Windows x86
Linux x64

**Mode:**
windows x64:
    set COMPlus_TieredCompilation=0
    set COMPlus_GCStress=0xC
    set COMPlus_ReadyToRun=0
    set COMPlus_ZapDisable=1
windows x86:
    set COMPlus_TieredCompilation=0
    set COMPlus_GCStress=0xC
    set COMPlus_JitStress=2
    set COMPlus_ReadyToRun=0
    set COMPlus_ZapDisable=1
Linux x64:
    export COMPlus_TieredCompilation=0
    export COMPlus_GCStress=0xC
    export COMPlus_ReadyToRun=0
    export COMPlus_ZapDisable=1

**Note:**
This failure **repro on** our windows x64 and Linux x64 machines but **doesn't repro** on windows x86. 

**Log:**
```
tracing/eventpipe/eventsvalidation/GCEvents/GCEvents.sh [FAIL]
      
      Return code:      1
      Raw output file:      /home/helixbot/work/95ccd71a-b8b9-444c-976e-d0058df23782/Work/5257efa6-45e9-4c67-969d-4fa1cf880b71/Exec/tracing/eventpipe/Reports/tracing.eventpipe/eventsvalidation/GCEvents/GCEvents.output.txt
      Raw output:
      BEGIN EXECUTION
      /home/helixbot/work/95ccd71a-b8b9-444c-976e-d0058df23782/Payload/corerun GCEvents.dll ''
        0.0s: ==TEST STARTING==
        0.9s: Connecting to EventPipe...
        9.5s: Connected to EventPipe with sessionID '0x7fb8c4000f60'
       10.2s: Started sending sentinel events...
       10.2s: Creating EventPipeEventSource...
       12.1s: EventPipeEventSource created
       13.0s: Dynamic.All callback registered
       13.0s: Running optional trace validator
       23.3s: Finished running optional trace validator
       23.3s: Starting stream processing...
       25.1s: Saw new provider 'Microsoft-Windows-DotNETRuntime'
       25.2s: Saw new provider 'Microsoft-DotNETCore-SampleProfiler'
       27.7s: Saw sentinel event
       27.7s: Starting event generating action...
       27.7s: Stopped sending sentinel events
       27.7s: Called GC.Collect() 0 times...
       27.8s: Called GC.Collect() 100 times...
       27.9s: Called GC.Collect() 200 times...
       28.0s: Called GC.Collect() 300 times...
       28.1s: Called GC.Collect() 400 times...
       28.2s: Called GC.Collect() 500 times...
       28.2s: Called GC.Collect() 600 times...
       28.3s: Called GC.Collect() 700 times...
       28.4s: Called GC.Collect() 800 times...
       28.5s: Called GC.Collect() 900 times...
       28.5s: Stopping event generating action
       28.5s: Sending StopTracing command...
       28.9s: Saw new provider 'Microsoft-DotNETCore-EventPipe'
       28.9s: Saw new provider 'Microsoft-Windows-DotNETRuntimeRundown'
       29.0s: Stopping stream processing
       29.0s: Dropped 0 events
       29.1s: Finished StopTracing command
       29.1s: Reader task finished
       29.1s: Validating optional callback...
       29.1s: Event counts validation
       29.1s: GCStartEvents: 13635
       29.1s: GCEndEvents: 13645
       29.1s: GCStartStopResult check: False
       29.1s: GCRestartEEStartEvents: 18510
       29.1s: GCRestartEEStopEvents: 18510
       29.1s: GCRestartEEStartStopResult check: True
       29.1s: GCSuspendEEEvents: 18418
       29.1s: GCSuspendEEEndEvents: 18417
       29.1s: GCSuspendEEStartStopResult check: True
       29.2s: ==TEST FINISHED: FAILED!==
      Expected: 100
      Actual: 255
      END EXECUTION - FAILED
      Test Harness Exitcode is : 1
```

**Comment by VincentBu** [comment]
when running tracing/eventpipe/eventsvalidation/GCEvents/GCEvents.sh on Linux x64, the console output is little different from the log on the helix.dot.net:
```
coreclr_initialize failed - status: 0x80004005
Expected: 100
Actual: 255
END EXECUTION - FAILED
```

**Comment by sywhang** [comment]
@josalem

**Comment by josalem** [comment]
Looks like the same failure as #26592.  I was able to reproduce this locally on my Windows x64 machine.  Taking a look at the test code, it looks like the test is looking for `bool GCStartStopResult = GCStartEvents >= 1000 && GCEndEvents >= 1000 && GCStartEvents == GCEndEvents;`, but I think this check is incorrect.  It is quite possible for up to two GCs to be in flight when the trace starts and stops, which means we might see deltas of 2.  The failure log above shows a delta of 10, which is rather odd.  I'm not sure that should be possible.  First thing we should do is correct the test to not check for equality, but rather `abs(start - end) <= 2`.  I haven't seen the delta greater than 2 reproduced locally, but I'll keep an eye out.

**Comment by josalem** [comment]
resolved by a combination of #26606 and #26741
